### ISSUE TITLE
Linux build --subsetCategory CoreCLR broken on x64 host

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/runtime#34292]
From the root directory, a build of the tip is failing like this

```bash
./build.sh --subsetCategory CoreCLR
```

```
Build FAILED.

/home/stmaclea/git/runtime/src/coreclr/crossgen-corelib.proj(17,5): error MSB3073: The command ""/home/stmaclea/git/runtime/src/coreclr/crossgen-corelib.sh" -debug" exited with code 253.
/home/stmaclea/git/runtime/.dotnet/sdk/5.0.100-preview.3.20170.6/Microsoft.Common.CurrentVersion.targets(4643,5): error MSB3030: Could not copy the file "/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/libjitinterface.so" because it was not found. [/home/stmaclea/git/runtime/src/coreclr/src/tools/crossgen2/crossgen2/crossgen2.csproj]
/home/stmaclea/git/runtime/.dotnet/sdk/5.0.100-preview.3.20170.6/Microsoft.Common.CurrentVersion.targets(4643,5): error MSB3030: Could not copy the file "/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/libclrjit.so" because it was not found. [/home/stmaclea/git/runtime/src/coreclr/src/tools/crossgen2/crossgen2/crossgen2.csproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error : Error when creating nuget lib package from /home/stmaclea/git/runtime/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.NETCore.ILDAsm.nuspec. NuGet.Packaging.Core.PackagingException: File not found: '/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/ildasm'. [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 254 [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error : Error when creating nuget packed package from /home/stmaclea/git/runtime/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.NETCore.ILDAsm.nuspec. NuGet.Packaging.Core.PackagingException: File not found: '/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/ildasm'. [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 254 [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error : Error when creating nuget packed package from /home/stmaclea/git/runtime/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.NETCore.ILDAsm.nuspec. NuGet.Packaging.Core.PackagingException: File not found: '/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/ildasm'. [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 254 [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error : Error when creating nuget lib package from /home/stmaclea/git/runtime/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.NETCore.TestHost.nuspec. NuGet.Packaging.Core.PackagingException: File not found: '/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/corerun'. [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 254 [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error : Error when creating nuget packed package from /home/stmaclea/git/runtime/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.NETCore.TestHost.nuspec. NuGet.Packaging.Core.PackagingException: File not found: '/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/corerun'. [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 254 [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error : Error when creating nuget packed package from /home/stmaclea/git/runtime/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.NETCore.TestHost.nuspec. NuGet.Packaging.Core.PackagingException: File not found: '/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/corerun'. [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 254 [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error : Error when creating nuget lib package from /home/stmaclea/git/runtime/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.NETCore.ILAsm.nuspec. NuGet.Packaging.Core.PackagingException: File not found: '/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/ilasm'. [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 254 [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error : Error when creating nuget packed package from /home/stmaclea/git/runtime/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.NETCore.ILAsm.nuspec. NuGet.Packaging.Core.PackagingException: File not found: '/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/ilasm'. [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 254 [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error : Error when creating nuget packed package from /home/stmaclea/git/runtime/artifacts/packages/Debug/specs/runtime.linux-x64.Microsoft.NETCore.ILAsm.nuspec. NuGet.Packaging.Core.PackagingException: File not found: '/home/stmaclea/git/runtime/artifacts/bin/coreclr/Linux..Debug/ilasm'. [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files) [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
/home/stmaclea/.nuget/packages/microsoft.dotnet.build.tasks.packaging/5.0.0-beta.20171.1/build/Packaging.targets(1260,5): error :    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Pack(String nuspecPath, String nupkgPath, Manifest manifest, Boolean packSymbols) in /_/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs:line 254 [/home/stmaclea/git/runtime/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
    0 Warning(s)
    12 Error(s)

```

**Comment by sdmaclea** [comment]
Also fails w/o `--subsetCategory`...

**Comment by jkoritzinsky** [comment]
@jkotas this might be related to your changes from ArchType to TargetArchitecture.

**Comment by AndyAyersMS** [comment]
If you pass `--arch x64` it still works.

**Comment by jkotas** [comment]
Taking a look...

**Comment by AndyAyersMS** [comment]
w/o --arch the build fails to find SPC for crossgen; I think it ends up in the `Linux..Release` directory
```

  Commencing Crossgenning System.Private.CoreLib
  __DistroRid: linux-x64
  __RuntimeId: linux-x64
  Setting up directories for build
  mkdir: cannot create directory ''

  /home/andy/repos/runtime/artifacts/bin/coreclr:
  total used in directory 32 available 38144036
  drwxr-xr-x   4 andy andy  4096 Mar 30 10:28 .
  drwxr-xr-x 274 andy andy 20480 Mar 30 10:59 ..
  drwxr-xr-x   5 andy andy  4096 Mar 30 10:38 Linux..Release
  drwxr-xr-x   6 andy andy  4096 Mar 30 10:38 Linux.x64.Release
```

**Comment by sdmaclea** [comment]
Strangely this seems to fail also

```bash
./build.sh --subsetCategory CoreCLR --arch x64 --configuration Release
```
